src/tools/rbd/action/Kernel.cc

```diff
@@ -188,6 +221,10 @@ static int do_kernel_map(const char *poolname, const char *imgname,
 
   free(devnode);
 out:
+  if (r < 0) {
+    print_error_description(poolname, imgname, snapname, -r);
```

Comment:
One reason for that is if this routine expands later with possibly other krbd calls, calling `print_error_description()` at a common exit point is neat.